jetbrains-core/src/software/aws/toolkits/jetbrains/ui/wizard/SamProjectGenerator.kt

```diff
@@ -108,10 +108,15 @@ class SamProjectGeneratorSettingsPeer(private val generator: SamProjectGenerator
     private val basePanel = SamInitSelectionPanel(settings)
     val sdkPanel: SdkSelectionPanelImpl by lazy { SdkSelectionPanelImpl(generator) }
 
-    // need a listener to autorun this? this hook is used in PyCharm
+    /**
+     * This hook is used in PyCharm and is called via {@link SamProjectBuilder#modifySettingsStep} for IntelliJ
+     */
     override fun validate(): ValidationInfo? {
-        val validationErrors = sdkPanel.validateAll()
-        return validationErrors?.firstOrNull()
+        val validationErrors = mutableListOf<ValidationInfo>()
+        basePanel.validate()?.let { validationErrors.add(0, it) }
```

Comment:
Index isn't needed

Reply:
Actually use `return basePanel.valudate() ?: sdkPanel.validateAll().firstOrNull()` so we don't need to do any memory allocations/copies